### DISCUSSION TITLE
Issue with the use of environment variables

### DISCUSSION THREAD

**Discussion by charlie45000** [davatorium/rofi#1464] [(Possible) Issues]
Hello,
I'm trying to switch from Xresources to rasi configuration and want to use environment variables. 

I had some issues with it:
- It seems that it isn't possible to use all environment variables (for example path doesn't work)
- variables are not convertible to strings, so properties that expect a string fail to load it (it is a big problem with cache-dir)
- I don't know if it is possible to combine string and environment variables (for exemple `$XDG_CACHE_HOME/rofi`)

I don't know it it is a bug or if it is simply not possible at the moment (in this case, would it be possible to add it?)

Thank you for your help!

**Comment by DaveDavenport** [comment]
For strings your environment variable needs to contain `"qoutes"` around it, but should work.
Currently the env needs to contain a valid full property. If there is interest I will extend the support and see if I can add f.e. string appending.

Currently environment are parsed as followed

```css
my {
aap : ${test}; 
}
```
or in 1.7.1:
```css
my {
aap: env(test, "default");
}
```

Then  test is parsed as the property value.  Whatever you can have between `:` and  `;` normally can be in the env.

**Reply by infokiller** [reply]
@DaveDavenport I'm trying to use an environment variable for `cache-dir` but it doesn't work with "unexpected forward slash", presumably because rofi expands the environment variable verbatim and I need to use it as a string. I tried:

```
configuration {
	cache-dir: env(HOST_HIST_DIR, "~/.cache");
}
```

Also, how can I append another string to an environment variable? For example if I want to use `${HOST_HIST_DIR}/rofi` as the value, how can this be done?

Thanks!

**Reply by DaveDavenport** [reply]
appending of string is not possible.  (might be an idea to add, need to think about it.).

HOST_HIST_DIR indeed needs to be formatted as a string (so surrounded by `"`
So if you set the environemnet make sure it includes this.

```bash
export HOST_HIST_DIR=\"/tmp\"
```
might work.

This is kind of a limitation of the system.  The configuration lexer needs to know the data type, it does this based on the content.
